Cargo.toml

```diff
@@ -13,11 +13,15 @@ categories = ["algorithms", "mathematics", "science"]
 ndarray = { version = "0.15", features = ["approx"] }
 num-traits = "0.2.0"
 thiserror = "1"
-rand = "0.8"
+rand = { version = "0.8", optional=true }
+rand_xoshiro = { version = "0.6", optional=true }
+ndarray-rand = { version = "0.14", optional=true }
```

Comment:
These 2 can be dev-dependencies after you make the other suggested changes. I'd rather have non-optional dev dependencies than optional but unnecessary normal dependencies.

Reply:
`ndarray-rand` is a testing dependency, I would like to keep the Xoroshiro random number generator as default to simplify the API (it has no additional dependency beside itself)